app/src/main/java/net/skyscanner/backpack/demo/MainActivity.kt

```diff
@@ -41,21 +64,90 @@ class MainActivity : BpkBaseActivity() {
+          BpkTopNavBar(
+            navIcon = NavIcon.None,
+            title = stringResource(R.string.app_name),
+            actions = listOf(
+              IconAction(icon = BpkIcon.Settings, contentDescription = stringResource(R.string.settings_title), onClick = {
+                val intent = Intent(context, SettingsActivity::class.java)
+                context.startActivity(intent)
+              })
+            )
+          )
+          LazyColumn {
+            item {
+              ComponentsTitle(stringResource(R.string.tokens_title))
+            }
+            items(ComponentRegistry.TOKENS.values.toList()) {
+              ComponentItem(title = it.name, showComposeBadge = hasComposeNodes(item = it))
+            }
+            item {
+              ComponentsTitle(title = stringResource(R.string.components_title))
+            }
+            items(ComponentRegistry.COMPONENTS.values.toList()) {
+              ComponentItem(title = it.name, showComposeBadge = hasComposeNodes(item = it))
+            }
+          }
+        }
+      }
+    }
+  }
 
-    val toolbar = findViewById<Toolbar>(R.id.toolbar)
-    setSupportActionBar(toolbar)
-    toolbar.title = title
+  @Composable
+  fun ComponentItem(
+    title: String,
+    showComposeBadge: Boolean,
+    modifier: Modifier = Modifier,
+  ) {
+    val context = LocalContext.current
+    Column(
+      modifier = modifier
+        .clickable {
+          val intent = Intent(context, ComponentDetailActivity::class.java)
+          intent.putExtra(ComponentDetailFragment.ARG_ITEM_ID, title)
+          context.startActivity(intent)
+        }
+        .height(56.dp)
+        .fillMaxWidth()
+        .padding(horizontal = BpkDimension.Spacing.Lg),
+    ) {
 
-    val componentsList = findViewById<View>(R.id.componentsList) as RecyclerView
-    val allItems = mutableListOf<StoriesRecyclerViewAdapter.ListItem>()
-    allItems.add(HeaderItem("Tokens"))
-    allItems.addAll(ComponentRegistry.TOKENS.map { it.toStoryItem() })
-    allItems.add(HeaderItem("Components"))
-    allItems.addAll(ComponentRegistry.COMPONENTS.map { it.toStoryItem() })
+      Row(
+        modifier = Modifier.weight(1f),
+        verticalAlignment = Alignment.CenterVertically,
+      ) {
+        BpkText(text = title, style = BpkTheme.typography.bodyLongform)
+        if (showComposeBadge) {
+          BpkBadge(
+            text = stringResource(R.string.story_badge_compose),
+            type = BpkBadgeType.Success,
+            modifier = Modifier.padding(start = BpkDimension.Spacing.Base)
+          )
+        }
+      }
+      Divider()
+    }
+  }
 
-    componentsList.adapter = StoriesRecyclerViewAdapter(allItems)
-    componentsList.addItemDecoration(StoryItemDecoration(this))
+  @Composable
+  fun ComponentsTitle(title: String, modifier: Modifier = Modifier) {
+    BpkText(
+      text = title.uppercase(),
+      modifier = modifier
+        .fillMaxWidth()
+        .padding(
```

Comment:
Looking at the original layout we can probably simplify this to
```
vertical = BpkDimension.Spacing.Base,
horizontal = BpkDimension.Spacing.Lg
```

Reply:
Thanks, I've fixed this!